src/components/SnapshotDetails/tabs/SnapshotPipelineRunsTab.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import { Bullseye, Spinner } from '@patternfly/react-core';
+import { PipelineRunLabel } from '../../../consts/pipelinerun';
+import { usePipelineRuns } from '../../../hooks/usePipelineRuns';
+import { StatusBox } from '../../../shared/components/status-box/StatusBox';
+import { PipelineRunKind } from '../../../types';
+import PipelineRunEmptyState from '../../PipelineRunDetailsView/PipelineRunEmptyState';
+import SnapshotPipelineRunsList from './SnapshotPipelineRunsList';
+
+type SnapshotPipelineRunTabProps = {
+  commit?: string;
+  snapshotName: string;
+  applicationName: string;
+  namespace: string;
+  customFilter?: (plr: PipelineRunKind) => boolean;
+};
+const SnapshotPipelineRunTab: React.FC<SnapshotPipelineRunTabProps> = ({
+  snapshotName,
+  applicationName,
+  namespace,
+  customFilter,
+}) => {
+  const [pipelineRuns, loaded, LoadError, getNextPage] = usePipelineRuns(
+    namespace,
+    React.useMemo(
+      () => ({
+        selector: {
+          matchLabels: {
+            [PipelineRunLabel.APPLICATION]: applicationName,
+          },
+        },
+      }),
+      [applicationName],
+    ),
+  );
```

Comment:
updated the others? So, should we remove the getNextPage here, as this list will show two pipelineruns one build, one test which is related to the snapshot

Reply:
I think the only issue here is that the first page of PLRs might not contain the PLRs for the snapshot (after filtering the PLRs by the snapshot annotation). So, if the length is 0 and `getNextPage` is not undefined, we should call `getNextPage()` in order to continue searching rather than showing the empty state.

```
  React.useEffect(() => {
    if (loaded && SnapshotPipelineRuns.length === 0 && getNextPage) {
      getNextPage();
    }
  }, [getNextPage, loaded]);
```